src/expressions.md

```diff
@@ -84,9 +84,15 @@ The following expressions can create mutable lvalues:
 ### Temporary lifetimes
 
 When using an rvalue in most lvalue contexts, a temporary unnamed lvalue is
-created and used instead. The lifetime of temporary values is typically the
-innermost enclosing statement; the tail expression of a block is considered
-part of the statement that encloses the block.
+created and used instead, if not promoted to `'static`. Promotion of an
```

Comment:
I think we need to document the "dataflow-based" behavior, because now it's an integral part of the Rust semantics.